### PR TITLE
i#2717 Change the implementation of options. (#2748)

### PR DESCRIPTION
**Pull request by nilayvaish**
The pull request is mostly what was there last time.  There are three more commits that were required to fix the issues identified with the pull request from last time.

### CODE REVIEW

**Comment by fhahn** [comment]
run aarch64 tests

**Comment by fhahn** [comment]
Could you add a reference to the previous PR that got merged? IIRC it was reverted because of a AArch64 test failure. The AArch64 precommit tests should now cover this, so we should be good to go.

**Comment by fhahn** [comment]
run aarch64 tests

**Comment by fhahn** [comment]
run aarch64 tests

File: core/options.c [hunk 1]
```diff
@@ -135,27 +135,6 @@ typedef struct _option_t {
...
-{ command_line_option, offsetof(options_t, name), sizeof(type), \
-  OPTION_TYPE_##type, pcache, OPTION_MOD_##flag},
-
-# define OPTION_COMMAND(type, name, default_value, command_line_option, \
-                        statement, description, flag, pcache) /* nothing */
-
-/* Read only source for default internal option values and names. No lock needed
- * since never written.
- */
-static const option_t internal_option_traits[] = {
```
Comment (derekbruening) [inline core/options.c]: OK, so for debug build, internal option traits are in the regular traits array, and for release build they remain as they were, a literal from an enum, so there's no reason for this separate traits array.

**Pull request merged**